services/app-api/src/postgres/contractAndRates/proto_to_db_RateRevisions.ts

```diff
@@ -0,0 +1,144 @@
+                )
+            }
+        }
+
+        for (const rateInfo of formData.rateInfos) {
+            const rateID = revision.pkgID
+
+            const existingRate = await client.rateTable.findFirst({
+                where: { id: rateID },
+            })
+            let createdRate: RateTable
+            if (!existingRate) {
+                createdRate = await client.rateTable.create({
+                    data: {
+                        id: rateID,
+                        createdAt: new Date(),
+                        updatedAt: new Date(),
+                        stateCode: formData.stateCode,
+                        stateNumber: formData.stateNumber,
+                    },
+                })
+                results.rate = createdRate
+            } else {
+                createdRate = existingRate
+                console.warn(
+                    `Rate with ID ${rateID} already exists. Skipping...`
+                )
+            }
+
+            const existingRateRevision =
+                await client.rateRevisionTable.findFirst({
+                    where: {
+                        rateID: createdRate.id,
+                        rateType: rateInfo.rateType,
+                        rateCapitationType: rateInfo.rateCapitationType,
+                        rateDateStart: rateInfo.rateDateStart,
+                        rateDateEnd: rateInfo.rateDateEnd,
+                    },
+                })
+            if (existingRateRevision) {
+                console.warn(
+                    `Rate revision for rate ID ${createdRate.id} with similar properties already exists. Skipping...`
+                )
+                continue
+            }
+
+            const rateRevisionId = uuid()
+
+            const rateRevision: RateRevisionTable = {
+                id: rateRevisionId,
+                rateID: createdRate.id,
+                createdAt: new Date(),
+                updatedAt: new Date(),
+                unlockInfoID: null,
+                submitInfoID: submitInfoID,
+                amendmentEffectiveDateStart:
+                    rateInfo.rateAmendmentInfo?.effectiveDateStart ?? null,
+                amendmentEffectiveDateEnd:
+                    rateInfo.rateAmendmentInfo?.effectiveDateEnd ?? null,
+                actuaryCommunicationPreference:
+                    rateInfo.actuaryCommunicationPreference ?? null,
+                rateType: rateInfo.rateType ?? null,
+                rateCapitationType: rateInfo.rateCapitationType ?? null,
+                rateDateStart: rateInfo.rateDateStart ?? null,
+                rateDateEnd: rateInfo.rateDateEnd ?? null,
+                rateDateCertified: rateInfo.rateDateCertified ?? null,
+                rateProgramIDs: rateInfo.rateProgramIDs ?? [],
+                rateCertificationName: rateInfo.rateCertificationName ?? null,
+            }
```

Comment:
This is missing `addtlActuaryContacts`. For this value, we need to get it from `formData.addtlActuaryContacts`. The `rateInfo` in `formData.rateInfos` does not have `addtlActuaryContacts` at all, but our new model does, so we just need to take it from the top level `formData`.

Reply:
There is also shared rates work that has not been done yet. So we should remember to address that once [MR-3568](https://qmacbis.atlassian.net/browse/MR-3568) is finished.